config.json

```diff
@@ -288,296 +307,395 @@
-        "slug": "clock",
-        "name": "Clock",
-        "uuid": "459fda78-851e-4bb0-a416-953528f46bd7",
-        "practices": [
-          "class-composition",
-          "rich-comparisons",
-          "string-formatting"
-        ],
-        "prerequisites": ["basics", "numbers", "strings", "classes"],
-        "difficulty": 4
-      },
-      {
-        "slug": "markdown",
-        "name": "Markdown",
-        "uuid": "88610b9a-6d3e-4924-a092-6d2f907ed4e2",
-        "practices": ["regular-expressions", "functions"],
-        "prerequisites": [],
-        "difficulty": 4
+        "slug": "collatz-conjecture",
+        "name": "Collatz Conjecture",
+        "uuid": "33f689ee-1d9c-4908-a71c-f84bff3510df",
+        "practices": ["numbers"],
+        "prerequisites": ["basics", "numbers"],
+        "difficulty": 1
       },
       {
-        "slug": "tournament",
-        "name": "Tournament",
-        "uuid": "49377a3f-38ba-4d61-b94c-a54cfc9034d0",
-        "practices": [
-          "anonymous-functions",
-          "collections",
-          "dicts",
-          "dict-methods",
-          "generators",
-          "generator-expressions",
-          "other-comprehensions",
-          "string-formatting",
-          "string-methods-splitting",
-          "tuples"
-        ],
+        "slug": "secret-handshake",
+        "name": "Secret Handshake",
+        "uuid": "0d5b2a0e-31ff-4c8c-a155-0406f7dca3ae",
+        "practices": ["list-methods"],
         "prerequisites": [
           "basics",
-          "strings",
-          "string-methods",
-          "dicts",
+          "bools",
+          "conditionals",
           "lists",
           "list-methods",
-          "loops",
-          "tuples"
+          "numbers",
+          "strings",
+          "string-methods"
         ],
-        "difficulty": 5
+        "difficulty": 1
       },
       {
-        "slug": "book-store",
-        "name": "Book Store",
-        "uuid": "4899b2ef-675f-4d14-b68a-1a457de91276",
-        "practices": [
-          "collections",
-          "conditionals",
-          "functools",
-          "generator-expressions",
-          "other-comprehensions",
-          "sets"
-        ],
+        "slug": "wordy",
+        "name": "Wordy",
+        "uuid": "af50bb9a-e400-49ce-966f-016c31720be1",
+        "practices": ["string-methods"],
         "prerequisites": [
           "basics",
-          "conditionals",
-          "dicts",
           "lists",
-          "list-methods",
           "loops",
-          "tuples",
-          "sets",
+          "strings",
+          "string-methods",
           "numbers"
         ],
-        "difficulty": 5
+        "difficulty": 1
       },
       {
-        "slug": "pangram",
-        "name": "Pangram",
-        "uuid": "bebf7ae6-1c35-48bc-926b-e053a975eb10",
-        "practices": ["sets"],
+        "slug": "triangle",
+        "name": "Triangle",
+        "uuid": "f0bc144f-3226-4e53-93ee-e60316b29e31",
+        "practices": ["bools"],
```

Comment:
The "restriction" is around how unlocking logic works.  If I have this exercise practice "bools", it will show up in the tree under "bools", and be unlocked by it.  Most students will then take on the exercise once they finish the "bools" concept exercise.  By the time they get to doing `sets`, this exercise will have already been "practiced", and so be redundant for them...and vice-versa.  If I have it practice `sets`, it will not be locked until the prerequisite is satisfied.  But that closes off the exercise from students who've only completed "bools".

....But maybe this is an an opportunity to create a similar problem or extended problem, purpose-built for a set-wise solution?  Analogous to "Resistor Color Duo" - only in this case, comparing triangles??   I don't know.  But the upshot (_at least for the early exercises_) is that we need to pick a "main" concept that an exercise practices.

Reply:
I am going to leave this "unresolved", just to have the convo open.  I think for now I will leave this as-is, with the thought we'd make extended or variant exercises to `triangle`.